flow/layers/container_layer.cc

```diff
@@ -6,13 +6,42 @@
 
 namespace flutter {
 
-ContainerLayer::ContainerLayer() {}
+ContainerLayer::ContainerLayer()
+    : child_needs_screen_readback_(false), renders_to_save_layer_(false) {}
 
 ContainerLayer::~ContainerLayer() = default;
 
 void ContainerLayer::Add(std::shared_ptr<Layer> layer) {
-  layer->set_parent(this);
+  Layer* the_layer = layer.get();
+  the_layer->set_parent(this);
   layers_.push_back(std::move(layer));
+  if (the_layer->tree_reads_surface()) {
+    update_child_readback(the_layer);
+  }
+}
+
+void ContainerLayer::update_child_readback(Layer* layer) {
```

Comment:
Nit: Here and elsewhere, please follow the [style guide on function names](https://google.github.io/styleguide/cppguide.html#Function_Names).

Reply:
If I'm reading the guildelines correctly, that would mean that the following should be moved to mixed case (and probably be off-line:

update_child_readback() => UpdateChildReadback()
update_screen_readback() => UpdateScreenReadback()
compute_tree_reads_surface() => ComputeTreeReadsSurface()

but all of the set_foo() and foo() methods for the properties themselves should continue to use variable naming style - even if the set_foo() methods call UpdateScreenReadback() as a side effect since their main use is to set the named property.

Does that match your expectation?

Reply:
Your reading is also my understanding: camel cases for all functions except the simple getters and setters.

Reply:
Yeah, that matches my understanding of what the style guide prescribes. 

> even if the set_foo() methods call UpdateScreenReadback() as a side effect since their main use is to set the named property.

Personally, I have always found this hard to determine and it is one of my gripes about the style guide and its lack of consistency. Either way is fine I suppose. But, if the function is more than a couple of lines, lets move it OOL and and use mixed case.

Reply:
set_foo() is semantically a setter, though. Should it really change to mixed case just because it is more than a couple of lines? Value propagation on setters is not an obscure concept, so I still think of it as just a setter.

Reply:
Let me know if you still have feedback on the naming/inlining used in the latest commit. I'll close this particular comment thread as the specific function it references has changed both name and inline status.